src/arrayMethodSort.js

```diff
@@ -4,8 +4,39 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const defaultCompareFunction = (a, b) => {
+      const strA = String(a);
+      const strB = String(b);
+
+      if (strA < strB) {
+        return -1;
+      }
+
+      if (strA > strB) {
+        return 1;
+      }
+
+      return 0;
+    };
+
+    let compare;
+
+    if (typeof compareFunction === 'function') {
+      compare = compareFunction;
+    } else {
+      compare = defaultCompareFunction;
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      for (let j = 0; j < this.length - 1; j++) {
+        if (compare(this[j], this[j + 1]) > 0) {
+          [this[j], this[j + 1]] = [this[j + 1], this[j]];
+        }
+      }
```

Comment:
In the inner loop, you can optimize the number of iterations by reducing the array length each time because the largest element will bubble up to the end of the array in each iteration. Change the condition to `j < this.length - i - 1`.